sdk/ai/azure-ai-projects/samples/agents/sample_agents_image_input_base64.py

```diff
@@ -0,0 +1,113 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use basic agent operations using image file input for the
+    the Azure Agents service using a synchronous client.
+
+USAGE:
+    python sample_agents_image_input_file.py
+
+    Before running the sample:
+
+    pip install azure-ai-projects azure-identity
+
+    Set these environment variables with your own values:
+    1) PROJECT_CONNECTION_STRING - The project connection string, as found in the overview page of your
+       Azure AI Foundry project.
+    2) MODEL_DEPLOYMENT_NAME - The deployment name of the AI model, as found under the "Name" column in 
+       the "Models + endpoints" tab in your Azure AI Foundry project.
+"""
+
+import os, time, base64
+from typing import List, Dict, Any, cast
+from azure.ai.projects import AIProjectClient
+from azure.identity import DefaultAzureCredential
+from azure.ai.projects.models import MessageTextContent, MessageContentBlockInput
+
+
+def image_to_base64(image_path: str) -> str:
+    """
+    Convert an image file to a Base64-encoded string.
+
+    :param image_path: The path to the image file (e.g. 'image_file.png')
+    :return: A Base64-encoded string representing the image.
+    :raises FileNotFoundError: If the provided file path does not exist.
+    :raises OSError: If there's an error reading the file.
+    """
+    if not os.path.isfile(image_path):
+        raise FileNotFoundError(f"File not found at: {image_path}")
+
+    try:
+        with open(image_path, "rb") as image_file:
+            file_data = image_file.read()
+        return base64.b64encode(file_data).decode("utf-8")
+    except Exception as exc:
+        raise OSError(f"Error reading file '{image_path}'") from exc
+
+
+# [START create_project_client]
+project_client = AIProjectClient.from_connection_string(
+    credential=DefaultAzureCredential(),
+    conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+)
+# [END create_project_client]
+
+with project_client:
+
+    # [START create_agent]
+    agent = project_client.agents.create_agent(
+        model=os.environ["MODEL_DEPLOYMENT_NAME"],
+        name="my-assistant",
+        instructions="You are helpful assistant",
+    )
+    # [END create_agent]
+    print(f"Created agent, agent ID: {agent.id}")
+
+    # [START create_thread]
+    thread = project_client.agents.create_thread()
+    # [END create_thread]
+    print(f"Created thread, thread ID: {thread.id}")
+
+    # [START create_message]
+    image_base64 = image_to_base64("image_file.png")
+    img_str = f"data:image/png;base64,{image_base64}"
+    input_message = "Hello, what is in the image ?"
+    content: List[Dict[str, Any]] = []
+    content = [{"type": "text", "text": input_message}]
+    content.append({"type": "image_url", "image_url": {"url": img_str, "detail": "high"}})
```

Comment:
Why not build the message list using strongly typed objects? You already have them defined.